integrated_tools_generator.py

```diff
@@ -0,0 +1,25 @@
+import itertools
+
+def interleave(*iterable):
+    """ get one iterable object or more and merge the objects contents """
+    permuts=[]
+    merged=[]
+    for item in iterable:
+     permut = itertools.permutations(item, len(item))
+     permuts.append(list(permut))
+    index1=0
```

Comment:
What are this indexes? you need to give all of your variables better names, the current names are making this code very hard for reading.